pkg/framer/property_value_history.go

```diff
@@ -12,79 +12,39 @@ import (
 )
 
 type AssetPropertyValueHistory struct {
-	*iotsitewise.BatchGetAssetPropertyValueHistoryOutput
+	*iotsitewise.GetAssetPropertyValueHistoryOutput
 	Query models.AssetPropertyValueQuery
 }
 
 func (p AssetPropertyValueHistory) Frames(ctx context.Context, resources resource.ResourceProvider) (data.Frames, error) {
-	frames := make(data.Frames, 0, len(p.SuccessEntries))
-	properties, err := resources.Properties(ctx)
-	if err != nil {
-		return frames, err
-	}
-
-	for _, h := range p.SuccessEntries {
-		frame, err := p.Frame(ctx, properties[*h.EntryId], h.AssetPropertyValueHistory)
-		if err != nil {
-			return nil, err
-		}
-		frames = append(frames, frame)
-	}
 
-	for _, e := range p.ErrorEntries {
-		property := properties[*e.EntryId]
-		frame := data.NewFrame(getFrameName(property))
-		if e.ErrorMessage != nil {
-			frame.Meta = &data.FrameMeta{
-				Notices: []data.Notice{{Severity: data.NoticeSeverityError, Text: *e.ErrorMessage}},
-			}
-		}
-		frames = append(frames, frame)
+	length := len(p.AssetPropertyValueHistory)
+	property, err := resources.Property(ctx)
+	if err != nil {
+		return nil, err
 	}
-
-	return frames, nil
-}
-
-func (p AssetPropertyValueHistory) Frame(ctx context.Context, property *iotsitewise.DescribeAssetPropertyOutput, h []*iotsitewise.AssetPropertyValue) (*data.Frame, error) {
-	length := len(h)
-
 	// TODO: make this work with the API instead of ad-hoc dataType inference
 	// https://github.com/grafana/iot-sitewise-datasource/issues/98#issuecomment-892947756
 	if *property.AssetProperty.DataType == *aws.String("?") {
-		if length != 0 {
-			property.AssetProperty.DataType = aws.String(getPropertyVariantValueType(h[0].Value))
-		} else {
-			property.AssetProperty.DataType = aws.String("")
-		}
+		property.AssetProperty.DataType = aws.String(getPropertyVariantValueType(p.AssetPropertyValueHistory[0].Value))
 	}
 
 	timeField := fields.TimeField(length)
 	valueField := fields.PropertyValueFieldForQuery(p.Query, property, length)
 	qualityField := fields.QualityField(length)
-	frameName := ""
-	if models.QueryTypePropertyAggregate == p.Query.QueryType {
-		frameName = getFrameName(property)
-	} else {
-		frameName = *property.AssetName
-	}
-	frame := data.NewFrame(
-		frameName,
-		timeField,
-		valueField,
-		qualityField)
-
+	frame := data.NewFrame(*property.AssetName, timeField, valueField, qualityField)
```

Comment:
Should we call `getFrameName()` here instead of *property.AssetName? I remember fixing this for "raw" queries, I assume it also applies for non-batch queries